npm_and_yarn/lib/dependabot/npm_and_yarn/file_parser.rb

```diff
@@ -36,7 +37,8 @@ def parse
         dependency_set = DependencySet.new
         dependency_set += manifest_dependencies
         dependency_set += lockfile_dependencies
-        dependencies = dependency_set.dependencies
+
+        dependencies = Helpers.dependencies_with_all_versions_metadata(dependency_set)
 
         # TODO: Currently, Dependabot can't handle dependencies that have both
         # a git source *and* a non-git source. Fix that!
```

Comment:
I think this is also an issue with the Terraform ecosystem because Dependencies using a Git source have a different internal identifier than Terraform registry dependencies.

Another thing that they may share in common now is, because npm and Terraform Dependencies can have multiple version requirements, the `requirements_changed?` method in Common needs to be overrided so that if one required version is already up-to-date, the `requirements_changed?` method still notices that the other required versions have been updated.

See https://github.com/dependabot/dependabot-core/pull/5786#issuecomment-1262533433 for more info

(this should be private)

```ruby
def requirement_changed?(file, dependency)
  changed_requirements =
    (dependency.requirements - dependency.previous_requirements) |
    (dependency.previous_requirements - dependency.requirements)

  changed_requirements.any? { |f| f[:file] == file.name }
end
```